src/vpn-api/worker.ts

```diff
@@ -143,13 +153,68 @@ const worker = (instanceId: number) => {
 			),
 		)
 		.tap(() => openvpn.start())
+		.delay(1000)
+		.tap(() => {
+			return vpn.connect({ port: mgtPort, shellPrompt: '' }).then(() => {
+				vpn.connection.shellPrompt = '';
+				// monitor new client connections and map cid to uuid
+				vpn.on('log', data => {
+					if (data.includes('CLIENT:ESTABLISHED,')) {
+						const clientId = data.split(',')[1].trim();
+						const idMapper = (logData: string) => {
+							if (logData.includes('CLIENT:ENV,common_name=')) {
+								const uuid = logData.split('=')[1].trim();
+								// expire any previous mappings for this device
+								_.forOwn(clientMap, (client, key) => {
```

Comment:
I've got a feeling this could be a performance issue in prod, I think it might be worth having two maps one clientId -> uuid and one uuid -> clientId so that you can change this to
```ts
delete clientMap[uuidMap[uuid]]
```
so that it becomes a constant time and also avoid the potentially dangerous delete on an object you're iterating (no idea if it is dangerous, but it's definitely something that often is).

Reply:
Done, I must admit, I had similar concerns.